src/lib.rs

```diff
@@ -173,6 +180,52 @@ pub mod graph {
 #[cfg(feature = "stable_graph")]
 pub use crate::graph_impl::stable_graph;
 
+
+pub mod lib {
+    #[cfg(not(feature = "std"))]
+    pub use alloc::string::{String, ToString};
+    #[cfg(feature = "std")]
+    pub use std::string::{String, ToString};
+
+    #[cfg(not(feature = "std"))]
+    pub use alloc::vec::Vec;
+    #[cfg(not(feature = "std"))]
+    pub use alloc::{format, vec};
+    #[cfg(feature = "std")]
+    pub use std::vec::{Vec};
+
+    #[cfg(not(feature = "std"))]
+    pub use alloc::boxed::Box;
+    #[cfg(feature = "std")]
+    pub use std::boxed::Box;
+
+    #[cfg(not(feature = "std"))]
+    pub use alloc::rc::Rc;
+    #[cfg(feature = "std")]
+    pub use std::rc::Rc;
+
+    #[cfg(not(feature = "std"))]
+    pub use alloc::sync::{Arc, Weak};
+    #[cfg(feature = "std")]
+    pub use std::sync::Arc;
+
+    #[cfg(feature = "std")]
+    pub use std::collections::hash_map::RandomState;
+    #[cfg(not(feature = "std"))]
+    pub use hashbrown::hash_map::DefaultHashBuilder as RandomState;
+
+    #[cfg(feature = "std")]
+    pub use std::collections::{hash_map::Entry::{Occupied, Vacant}, HashSet, hash_map::Iter};
+    #[cfg(not(feature = "std"))]
+    pub use hashbrown::{ HashSet, hash_map::{ HashMap, Entry::{Occupied, Vacant}, Iter}};
```

Comment:
HashMap is used in the public API - this PR would mean that public API changes breakingly based on a cargo feature, and that's not an allowed usage of cargo features in general.